client/src/view/actions-panel.tsx

```diff
@@ -250,7 +257,11 @@ export class ActionsPanel extends React.Component<IActionsPanelProps, object> {
                 outline={true}
                 color="primary"
                 className="command-button"
-                onClick={() => this.props.islandState.stateAfterCommand(command, this.props.location).dispatch()}
+                onClick={() => {
+                    const props = this.props
+                    const commandHandler = new CommandHandler(props.appState, props.stateSubject)
+                    props.stateSubject.next(commandHandler.afterCommand(command, props.location))
+                }}
```

Comment:
Should really extract this to keep the JSX simple

Reply:
The JSX here is just a Button, so I think it can handle some clutter so it's self-contained.